### PR TITLE
Visu samples

### LINKED ISSUE #2090: QA Day: Samples with visualization
Chekc that samples with visualization display as expected

### LINKED ISSUE #2116: make assert_features(list) raise an exception if a feature is not in list
currently assert_features does not raise an error:
run e.g. doc/tutorials/06-active_matter/SOLUTIONS/enhanced_diffusion.py without the engine feature compiled in (assert_features() does not raise an exception here)

### PR DESCRIPTION
**Pull request by KonradBreitsprecher**
Fixes #2090. Fixes #2116.

- Fixed a bug in the p3m init order that caused a segfault if the skin was set before p3m.

- Tested and added docstring and feature-check for:

visualization_bonded.py		
visualization_cellsystem.py	
visualization_charged.py	
visualization_constraints.py	
visualization_interactive.py	
visualization_lbboundaries.py   
visualization.py -> visualization_ljliquid.py	
visualization_mmm2d.py		
visualization_npt.py		
visualization_poisseuille.py	
billard.py
chamber_game.py
drude_bmimpf6.py

### CODE REVIEW

File: samples/lj_liquid_distribution.py [hunk 1]
```diff
@@ -19,8 +19,9 @@
 from __future__ import print_function
 import numpy as np
 import espressomd
+from espressomd import visualization
 from espressomd import thermostat
-from samples_common import open
+#from samples_common import open
```
Comment (fweik) [inline samples/lj_liquid_distribution.py]: Comment

File: samples/lj_liquid_distribution.py [hunk 2]
```diff
@@ -199,6 +203,8 @@
     r, dist = system.analysis.distribution(type_list_a=distr_type_list_a, type_list_b=distr_type_list_b,
                                            r_min=distr_r_min, r_max=distr_r_max, r_bins=distr_r_bins,
                                            log_flag=distr_log_flag, int_flag=distr_int_flag)
+    print(dist)
```
Comment (fweik) [inline samples/lj_liquid_distribution.py]: Debug stuff

**Review by fweik — CHANGES_REQUESTED** [review]

File: samples/lj_liquid_distribution.py (hunk 2 above)
Comment (KonradBreitsprecher) [reply samples/lj_liquid_distribution.py]: Revoked changes there, thanks for spotting...

**Comment by KonradBreitsprecher** [comment]
I found that `p3m_calc_differential_operator()` was in the wrong spot.
It crashed for manual p3m parameter setting at first use of `d_op` in `calc_influence_function_force()`.
It worked with tuning, probably because of returns in the init routine.

Ok ce781438d0db7a13d023b4799ffee6aaf1430b46 did this already..

File: src/core/p3m.cpp [hunk 3]
```diff
@@ -1207,12 +1207,7 @@ template <int cao> void calc_influence_function_force() {
                  Utils::sqr(p3m.d_op[RY][n[KY]] / box_l[RY]) +
                  Utils::sqr(p3m.d_op[RZ][n[KZ]] / box_l[RZ]);
 
-          if(fak2 == 0) {
-            fak3 = 0;
-          }
-          else
-            fak3 = fak1 / (fak2 * Utils::sqr(denominator));
-
```
Comment (mkuron) [inline src/core/p3m.cpp]: Looks like you made an error resolving a merge conflict. The check for zero was intentionally added in #2130.

**Review by mkuron — CHANGES_REQUESTED** [review]

**Comment by KonradBreitsprecher** [comment]
Thx @mkuron somehow missed that.

**Review by fweik — APPROVED** [review]

**Review by fweik — APPROVED** [review]

**Pull request merged**